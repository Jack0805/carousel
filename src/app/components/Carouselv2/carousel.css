.carouselv2-container {
    overflow: hidden;
    position: relative;
    width: 100vw;
    height: 100vh;
    margin: 0 auto;
    padding: 20px;
}

.carouselv2-track {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap:10px;
    transition: transform 1s ease;
    max-width: 100vw
}

.carouselv2-item {
    min-width: 260px;
    height: 390px;
    background: #606060;
}

.carouselv2-item-placehodler{
    min-width: 130px;
    background: black;
}

.c-img {
    width: 260px;
    opacity: 0;
    transition: opacity 0.4s ease-in-out;
}

.loaded{
    opacity: 1;
}