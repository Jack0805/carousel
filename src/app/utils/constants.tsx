export const endPoint = '/data/data.json'