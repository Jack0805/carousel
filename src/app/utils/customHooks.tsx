'use client'
import {useState, useEffect, useRef} from 'react';

export function useViewportWidth() {
    const isBrowser = typeof window !== 'undefined';
    const [viewportWidth, setViewportWidth] = useState(isBrowser ? window.innerWidth : 0);

    useEffect(() => {

        if (!isBrowser) {
            return;
        }
        const handleResize = () => {
            setViewportWidth(window.innerWidth);
        };

        window.addEventListener('resize', handleResize);
        return () => {
            window.removeEventListener('resize', handleResize);
        };
    }, [isBrowser]);

    switch (true) {
        case viewportWidth > 1600:
            return 6;
        case viewportWidth > 1080:
            return 5;
        case viewportWidth > 720:
            return 4;
        default:
            return 3;
    }
}

interface UseAutoFocusState {
    index:number,
}
export function useAutoFocus({index}: UseAutoFocusState) {

    const myDivRef = useRef<HTMLDivElement>(null);

    useEffect(() => {

        if (myDivRef.current && index == 0) {
            myDivRef.current.focus();
        }
        const handleClickOutside = (event: MouseEvent) => {
            if (myDivRef.current && !myDivRef.current.contains(event.target as Node) && index == 0) {
                myDivRef.current.focus();
            }
        };

        document.addEventListener('click', handleClickOutside);

        return () => {
            document.removeEventListener('click', handleClickOutside);
        };

    }, []);

    return {
        myDivRef
    }
}