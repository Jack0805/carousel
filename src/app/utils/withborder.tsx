import React, {ComponentType, useRef, useEffect} from 'react';
import {movieProps} from "@/app/components/Carousel/movie";

const withBorder = <P extends movieProps>(Component: ComponentType<P>) => {
    return (props: P)  => (
        <div
            className={props.selected ? 'image-container-with-border':'image-container-without-border'}
        >
            <Component {...props}/>
        </div>
    );
};

export default withBorder;