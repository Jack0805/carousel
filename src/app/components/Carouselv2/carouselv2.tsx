'use client'
import React, {useEffect, useRef, useState, useCallback, KeyboardEvent, SetStateAction} from 'react';
import './carousel.css';
import {useAppDispatch,useAppSelector} from "@/app/state/hooks";
import {RootState} from "@/app/state/store";
import {fetchMoviesThunk, getNextMoviesOnScreenArr, getPreviousMoviesOnScreenArr} from "@/app/state/movies/movieSlice";
import {CarouselProps} from "../Carousel/carousel"
import {useRouter} from "next/navigation";
import Moviev2 from "@/app/components/Carouselv2/moviev2";


function Carouselv2({endPoint,maxNumberOfItem}:CarouselProps) {
    const [currentPosition, setCurrentPosition] = useState(0)
    const [selectedIndex, setSelectedIndex] = useState<number>(0)
    const [triggerIndexRight,setTriggerIndexRight] = useState<number | null>(-1);
    const [triggerIndexLeft,setTriggerIndexLeft] = useState<number | null>(-1);
    const carouselRef = useRef<HTMLDivElement | null>(null);
    const dispatch = useAppDispatch();
    const movies = useAppSelector((state: RootState) => state.movies.movies);
    const router = useRouter();


    useEffect(() => {
        dispatch(fetchMoviesThunk({endPoint, maxNumberOfItem, errorMessage: 'An unknown error occurred, please try again later'}))
    }, []);

    useEffect(() => {

        const options = {
            root: document.getElementById('root'),
            rootMargin: '10px',
            threshold: 0.01
        };

        const observer = new IntersectionObserver((entries) => {
            entries.forEach((entry) => {
                if (entry.isIntersecting) {
                    if (entry.intersectionRect.width < entry.boundingClientRect.width && entry.boundingClientRect.right > window.innerWidth) {
                        const dataKey = entry.target.getAttribute('data-key') as SetStateAction<any>;
                        setTriggerIndexRight(parseInt(dataKey))
                    } else if (entry.intersectionRect.width < entry.boundingClientRect.width && entry.boundingClientRect.left < 0) {
                        const dataKey = entry.target.getAttribute('data-key') as SetStateAction<any>;
                        setTriggerIndexLeft(parseInt(dataKey))
                    } else if (entry.target.getAttribute('data-key') as SetStateAction<any> == movies.length -1 ){
                        setTriggerIndexRight(-1)
                    }

                    const imgElement = entry.target.querySelector('img');
                    if (imgElement) {
                        imgElement.setAttribute('src', entry.target.getAttribute('data-src') as string)
                        imgElement.setAttribute('style', 'opacity: 1;');
                    }

                }
                if (!entry.isIntersecting) {
                    if (entry.intersectionRect.width > 0 && entry.intersectionRect.width < entry.boundingClientRect.width && entry.boundingClientRect.left < 0) {
                        const dataKey = entry.target.getAttribute('data-key') as SetStateAction<any>;
                        setTriggerIndexLeft(parseInt(dataKey))
                    }
                    const imgElement = entry.target.querySelector('img');
                    if (imgElement) {
                        imgElement.setAttribute('src', '')
                        imgElement.setAttribute('style', 'opacity: 0;transition: opacity 1s ease-in-out;')
                    }
                }
            })
        },options)

        const images = document.querySelectorAll(`.carouselv2-item`);

        images.forEach((img) => observer.observe(img))

        return () => {
            observer.disconnect()
        }

    }, [movies]);

    const handleKeyPress = (event: KeyboardEvent<HTMLDivElement>) => {
        if (event.key === 'ArrowLeft') {
            setTimeout(() => {
                if (selectedIndex != 0) {
                    if ((selectedIndex-1) == triggerIndexLeft) {
                        setSelectedIndex(selectedIndex - 1)
                        setCurrentPosition(((prePosition) => (prePosition - 0.85)));
                        setTriggerIndexRight(triggerIndexLeft + 1)
                    } else {
                        setSelectedIndex(selectedIndex - 1)
                    }
                }
            },150)
        } else if (event.key === 'ArrowRight') {
            setTimeout(() => {
                if ((selectedIndex + 1) != movies.length) {
                    if ((selectedIndex+1) == triggerIndexRight) {
                        setSelectedIndex(selectedIndex + 1)
                        setCurrentPosition(((prePosition) => (prePosition + 0.85)));
                        setTriggerIndexLeft(triggerIndexRight-1)
                    } else {
                        setSelectedIndex(selectedIndex + 1)
                    }
                }
            },100)
        } else if (event.key === 'Enter'){
            router.push(`/program/?mid=${movies[selectedIndex].id}`)
        }
    }

    return(
        <div className="carouselv2-container">
            <div className="carouselv2-track" ref={carouselRef} style={{transform: `translateX(${-currentPosition*100}%)`}}>
                <span className="carouselv2-item-placehodler" />
                {
                    movies.map((item, index) => {
                        return (
                            <Moviev2
                                movie={item}
                                key={index}
                                index={index}
                                selected={index===selectedIndex}
                                selectedIndex={index}
                                currentPage = {0}
                                handleClick = {()=>null}
                                handleKeyPress={handleKeyPress}
                            />
                        )
                    })
                }
            </div>
        </div>
    );
}

export default Carouselv2;