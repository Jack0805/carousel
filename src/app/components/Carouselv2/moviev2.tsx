import React from 'react';
import withBorder from '../../utils/withborder'
import Link from "next/link";
import {useAutoFocus} from '../../utils/customHooks';
import {movieProps} from '../Carousel/movie'

const Moviev2 = (props: movieProps) => {
    const {id, image} = props.movie
    const {myDivRef} = useAutoFocus({index:props.index || 0})
    return (
                <div className="carouselv2-item" data-key={props.index} data-src={image} onKeyDown={(e) => props.handleKeyPress(e)} tabIndex={0} ref={myDivRef} style={{ outline: 'none' }}>
                    <Link href={{ pathname: '/program', query: { mid:id, from:'tvshows' }}}><img className="c-img" /></Link>
                </div>
    );
};

export default withBorder(Moviev2);