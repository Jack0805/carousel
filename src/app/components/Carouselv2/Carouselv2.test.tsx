import { render,act } from '@testing-library/react';
import React from 'react';
import Carouselv2 from "@/app/components/Carouselv2/carouselv2";
import {endPoint} from "@/app/utils/constants";
import {Provider} from "react-redux";
import configureMockStore from 'redux-mock-store';
import '@testing-library/jest-dom';
import mockFetch from '../../utils/jest';
import thunk from 'redux-thunk';

jest.mock('next/navigation', () => ({
    useRouter: jest.fn()
}))
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Tv Shows Component', () => {
    it('should render correctly', async () => {
        const mockData = await mockFetch.get()
        class IntersectionObserverMock {
            observe = jest.fn();
            disconnect = jest.fn();
            constructor(callback: (entries: IntersectionObserverEntry[]) => void) {
                act(() => {
                    callback([
                        {
                            isIntersecting: true,
                            target: document.createElement('div'),
                            intersectionRect: {width: 100, height: 100},
                            boundingClientRect: {width: 200, height: 200, right: 300, left: 100},
                        } as unknown as IntersectionObserverEntry,
                    ]);
                });
            }
        }
        // @ts-ignore
        window.IntersectionObserver = IntersectionObserverMock;

        const store = mockStore({
            movies: {
                movies: mockData,
                moviesOnScreen: [],
                previousPageState: {
                    selectedIndex: 0,
                    pageNum: 1
                }
            },
        });
        const { container } = render(
            <Provider store={store}>
                <Carouselv2 endPoint={endPoint} maxNumberOfItem={6} />
            </Provider>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
})
