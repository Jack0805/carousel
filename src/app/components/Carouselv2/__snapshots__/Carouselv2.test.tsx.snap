// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tv Shows Component should render correctly 1`] = `
<div
  class="carouselv2-container"
>
  <div
    class="carouselv2-track"
    style="transform: translateX(0%);"
  >
    <span
      class="carouselv2-item-placehodler"
    />
    <div
      class="image-container-with-border"
    >
      <div
        class="carouselv2-item"
        data-key="0"
        data-src="https://streamcoimg-a.akamaihd.net/000/672/98/67298-PosterArt-2039396c9e27d6271c96776414d6a38c.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=67298&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="1"
        data-src="https://streamcoimg-a.akamaihd.net/000/657/37/65737-PosterArt-15bee119eb92a5f33670e6bd3f1af967.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=65737&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="2"
        data-src="https://streamcoimg-a.akamaihd.net/000/675/17/67517-PosterArt-d740bf2159cf9c3eb1b22db4043076ff.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=67517&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="3"
        data-src="https://streamcoimg-a.akamaihd.net/000/561/97/56197-PosterArt-469c02a0132ce2629859f70303083c85.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=56197&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="4"
        data-src="https://streamcoimg-a.akamaihd.net/000/267/02/26702-PosterArt-fa4764f2e5af823683130cec70f227ae.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=26702&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="5"
        data-src="https://streamcoimg-a.akamaihd.net/000/301/9990/3019990-PosterArt-5984e294f596ed0ad0c006162770e080.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3019990&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="6"
        data-src="https://streamcoimg-a.akamaihd.net/000/675/43/67543-PosterArt-44056c0edeaae6d3dcd6874e2939c2f6.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=67543&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="7"
        data-src="https://streamcoimg-a.akamaihd.net/000/675/42/67542-PosterArt-31c442575a5b2142e4f6f8bd4b27b635.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=67542&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="8"
        data-src="https://streamcoimg-a.akamaihd.net/000/601/50/60150-PosterArt-156ce8779e10cea3cdd77c34036a9c6f.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=60150&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="9"
        data-src="https://streamcoimg-a.akamaihd.net/000/299/3105/2993105-PosterArt-7005562eb1f40b180ae93a4c99606d76.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=2993105&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="10"
        data-src="https://streamcoimg-a.akamaihd.net/000/301/9366/3019366-PosterArt-c9fb9a46950f279aae6170eabab2da93.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3019366&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="11"
        data-src="https://streamcoimg-a.akamaihd.net/000/375/815/375815-PosterArt-db43c606be4873c5058a9b412a2a01db.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=375815&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="12"
        data-src="https://streamcoimg-a.akamaihd.net/000/637/30/63730-PosterArt-6ad42c53470b4ab69ddbbeaf23f2962e.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=63730&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="13"
        data-src="https://streamcoimg-a.akamaihd.net/000/199/71/19971-PosterArt-5005731d277e57919be404f2cb7bbb7b.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=19971&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="14"
        data-src="https://streamcoimg-a.akamaihd.net/000/207/89/20789-PosterArt-3f323c1819174a64666046c962ec3898.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=20789&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="15"
        data-src="https://streamcoimg-a.akamaihd.net/000/276/56/27656-PosterArt-3f5aaa3c3baf5495b200c980543ffc9f.jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=27656&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="16"
        data-src="https://streamcoimg-a.akamaihd.net/000/302/8275/3028275-PosterArt-3484d8b77cab2bdd7394aa2a8471b694.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3028275&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="17"
        data-src="https://streamcoimg-a.akamaihd.net/000/680/70/68070-PosterArt-9c14d0f6f6029b75d05f38e022f0c921.jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=68070&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="18"
        data-src="https://streamcoimg-a.akamaihd.net/000/667/76/66776-PosterArt-8bb9065627b00fc779caaab0531a0da3.jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=66776&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="19"
        data-src="https://streamcoimg-a.akamaihd.net/000/295/79/29579-PosterArt-02c4dff7a346a23df002e3434e190cdf.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=29579&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="20"
        data-src="https://streamcoimg-a.akamaihd.net/000/408/308/408308-PosterArt-e607fb9b06d070b055064b7b861e3341.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=408308&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="21"
        data-src="https://streamcoimg-a.akamaihd.net/000/302/0031/3020031-PosterArt-f73e3f1d66bda1d37ad0782cd366b115.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3020031&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="22"
        data-src="https://streamcoimg-a.akamaihd.net/000/302/8280/3028280-PosterArt-162b2fbc055243013eff73aae0f4de72.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3028280&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="23"
        data-src="https://streamcoimg-a.akamaihd.net/000/264/21/26421-PosterArt-21077e57c8e85b842997cea8deb3a8f1.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=26421&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="24"
        data-src="https://streamcoimg-a.akamaihd.net/000/681/48/68148-PosterArt-935533a8435c3700a4354752577718f0.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=68148&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="25"
        data-src="https://streamcoimg-a.akamaihd.net/000/301/9355/3019355-PosterArt-b99bca2ca0ef972600e4ab7cb414a228.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3019355&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="26"
        data-src="https://streamcoimg-a.akamaihd.net/000/676/63/67663-PosterArt-5fa8f418534239af025e67ac5ecc2aec.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=67663&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="27"
        data-src="https://streamcoimg-a.akamaihd.net/000/302/8281/3028281-PosterArt-27c7bf01c3e72dcf70b934bc06e7e29d.jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=3028281&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
    <div
      class="image-container-without-border"
    >
      <div
        class="carouselv2-item"
        data-key="28"
        data-src="https://streamcoimg-a.akamaihd.net/000/268/95/26895-PosterArt-b64690765414cbb82c81323f63b3588e.jpg?resize=512px:*&quality=75&preferredFormat=image/jpeg"
        style="outline: none;"
        tabindex="0"
      >
        <a
          href="/program?mid=26895&from=tvshows"
        >
          <img
            class="c-img"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
